nucypher/characters/lawful.py

```diff
@@ -835,6 +871,41 @@ def retrieve(self,
 
         return cleartexts
 
+    def matching_nodes_among(self,
+                             nodes: FleetSensor,
+                             no_less_than=7):  # Somewhat arbitrary floor here.
+        # Look for nodes whose checksum address has the second character of Bob's encrypting key in the first
+        # few characters.
+        # Think of it as a cheap knockoff hamming distance.
+        # The good news is that Bob can construct the list easily.
+        # And - famous last words incoming - there's no cognizable attack surface.
+        # Sure, Bob can mine encrypting keypairs until he gets the set of target Ursulas on which Alice can
+        # store a TreasureMap.  And then... ???... profit?
```

Comment:
This algorithm is kind of quirky and I'm not sure it works as the network grows. As far as I understand, the whole point is that Bob recovers roughly the same list of target nodes that Alice "sampled", and where the TMap is stored, correct?

Let's imagine a scenario where initially we have very few nodes. Alice creates a TMap, and since there are few nodes, the algorithm is forced to expand the search boundary a few times. @jMyles plays the Song of Time and we move 1 year to the future, where we have a bigger network; now Bob tries to retrieve the TMap, and since we have more nodes, there's greater probability that there are matches with small search boundaries. This has the effect that some of the initial target nodes are not discovered by Bob.

I don't think however this is a deal-breaker, but perhaps we should open an issue.

Reply:
Yes!  This is why I questioned the minimization of #2124 in the call yesterday.

Reply:
Glad that you already had an issue tracking this
```suggestion
        # store a TreasureMap.  And then... ???... profit?
        # TODO: There's some risk that Bob can't get any of the original target Ursulas - See #2124 
```